src/MongoDB/ReadPreference.c

```diff
@@ -471,6 +483,132 @@ static PHP_METHOD(ReadPreference, bsonSerialize)
 	convert_to_object(return_value);
 } /* }}} */
 
+/* {{{ proto string MongoDB\Driver\ReadPreference::serialize()
+*/
+static PHP_METHOD(ReadPreference, serialize)
+{
+	php_phongo_readpreference_t* intern;
+	ZVAL_RETVAL_TYPE             retval;
+	php_serialize_data_t         var_hash;
+	smart_str                    buf = { 0 };
+	const char*                  modeString = NULL;
+	const bson_t*                tags;
+	int64_t                      maxStalenessSeconds;
+	mongoc_read_mode_t           mode;
+
+	intern = Z_READPREFERENCE_OBJ_P(getThis());
+
+	if (zend_parse_parameters_none() == FAILURE) {
+		return;
+	}
+
+	if (!intern->read_preference) {
+		return;
+	}
+
+	tags                = mongoc_read_prefs_get_tags(intern->read_preference);
+	mode                = mongoc_read_prefs_get_mode(intern->read_preference);
+	modeString          = php_phongo_readpreference_get_mode_string(mode TSRMLS_CC);
+	maxStalenessSeconds = mongoc_read_prefs_get_max_staleness_seconds(intern->read_preference);
+
+#if PHP_VERSION_ID >= 70000
+	array_init_size(&retval, 3);
+#else
+	ALLOC_INIT_ZVAL(retval);
+	array_init_size(retval, 3);
+#endif
+
+	if (modeString) {
+#if PHP_VERSION_ID >= 70000
+		ADD_ASSOC_STRING(&retval, "mode", modeString);
+#else
+		ADD_ASSOC_STRING(retval, "mode", modeString);
+#endif
+	}
+
+	if (!bson_empty0(tags)) {
+		php_phongo_bson_state state = PHONGO_BSON_STATE_INITIALIZER;
+		/* Use native arrays for debugging output */
+		state.map.root_type     = PHONGO_TYPEMAP_NATIVE_ARRAY;
+		state.map.document_type = PHONGO_TYPEMAP_NATIVE_ARRAY;
```

Comment:
If #1030 is merged first, this can be rebased to use the new macro.

Reply:
I'd like to get this merged first before dealing with the macros and rebase #1030 instead.